packages/gestalt/src/SideNavigation/NavigationContent.js

```diff
@@ -0,0 +1,149 @@
+// @flow strict
+import { type Node as ReactNode, useEffect, useRef, useState } from 'react';
+import classnames from 'classnames';
+import Collapser from './Collapser';
+import {
+  countItemsWithIcon,
+  groupIconlessChildren,
+  validateChildren,
+} from './navigationChildrenUtils';
+import borderStyles from '../Borders.css';
+import Box from '../Box';
+import boxStyles from '../Box.css';
+import { useSideNavigation } from '../contexts/SideNavigationProvider';
+import Divider from '../Divider';
+import Flex from '../Flex';
+import layoutStyles from '../Layout.css';
+import { type Props as SideNavigationProps } from '../SideNavigation';
+import styles from '../SideNavigation.css';
+import { flattenChildrenWithKeys } from '../utils/flattenChildren';
+import { FixedZIndex } from '../zIndex';
+
+type Props = { ...SideNavigationProps };
+type Timeout = ReturnType<typeof setTimeout>;
+
+export default function NavigationContent({
+  accessibilityLabel,
+  children,
+  footer,
+  header,
+  showBorder,
+}: Props): ReactNode {
+  const navigationChildren = flattenChildrenWithKeys(children);
+
+  validateChildren({ children: navigationChildren, filterLevel: 'main' });
+
+  const {
+    collapsible,
+    collapsed: sideNavigationCollapsed,
+    onCollapse,
+    overlayPreview,
+    setOverlayPreview,
+    transitioning,
+  } = useSideNavigation();
+
+  const scrollContainer = useRef<HTMLDivElement | null>(null);
+  const [isScrolled, setIsScrolled] = useState(false);
+  const previewTimeoutRef = useRef<?Timeout>();
+
+  useEffect(() => {
+    const element = scrollContainer.current;
+    const scrollHandler = () => setIsScrolled(!!element?.scrollTop);
+
+    const mouseEnterHandler = () => {
+      if (sideNavigationCollapsed && !transitioning) {
+        clearTimeout(previewTimeoutRef.current);
```

Comment:
clearTimeout here is not really doing much right?
 

![Screenshot by Dropbox Capture](https://github.com/pinterest/gestalt/assets/10593890/b197b6e0-8565-4271-bd7a-de83cf526eb1)

Reply:
mouse can enter while sidenav is in preview mode. when that happens, we should remove the timer, so it doesn't close while the mouse is still on sidenavigation.